File: backend/api/app/domain/entities/message.py
import json
from dataclasses import dataclass

from app.domain.entities import BaseEntity
from pydantic import BaseModel


class Message(BaseModel):
    user_id: str
    payload: dict
    model: str
    isTeam: bool = False
    
    class Config:
        arbitrary_types_allowed = True

@dataclass
class MessageEntity(BaseEntity):
    id: str
    sender: str
    content: dict
    date: str
    
    @classmethod
    def from_dict(cls, data: dict) -> 'MessageEntity':
        return cls(
            id=data.get('id'),
            sender=data.get('sender'),
            content=data.get('content'),
            date=data.get('date')
        )
    
    def to_dict(self) -> dict:
        return {
            'id': self.id,
            'sender': self.sender,
            'content': self.content,
            'date': str(self.date)
        }
    
    def to_json(self) -> str:
        return json.dumps(self.to_dict())

    @classmethod
    def from_json(cls, json_str: str) -> 'MessageEntity':
        data = json.loads(json_str)
        return cls.from_dict(data)

File: mobile/lib/features/architect/data/models/message.dart
import '../../domains/entities/message.dart';

class MessageModel extends Message {
  const MessageModel({
    required String id,
    required String sender,
    required DateTime date,
    required String content,
  }) : super(
          id: id,
          sender: sender,
          date: date,
          content: content,
        );
  factory MessageModel.fromJson(Map<String, dynamic> json) {
    return MessageModel(
        id: json['id'],
        sender: json['sender'],
        date: DateTime.parse(json['date']),
        content: json['content']);
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'sender': sender,
      'date': date,
      'content': content,
    };
  }
}
